tests/TestingTools/TaskBuilder.ts

```diff
@@ -0,0 +1,163 @@
+// Builder
+import type { Moment } from 'moment';
+import { Priority, Status, Task } from '../../src/Task';
+import type { Recurrence } from '../../src/Recurrence';
+import { DateParser } from '../../src/Query/DateParser';
+
+/**
+ * A fluent class for creating tasks for tests.
+ *
+ * This uses the Builder Pattern.
+ *
+ * See TaskBuilder.build() for an example of use.
+ *
+ * IMPORTANT: Changed values are retained after calls to .build()
+ *            There is no way to reset a TaskBuilder to its default
+ *            start currently.
+ *            Create a new TaskBuilder object to start from a clean state,
+ */
+export class TaskBuilder {
+    private _status: Status = Status.Todo;
+    private _description: string = 'my description';
```

Comment:
Is there a set restricting on description, max length, expected characters/emoji? If so set it to a complex string with all the expected properties in it?

Reply:
That's a good question. I don't know the answer to that. I copied the types from the actual Task class, so I doubt there is a restriction in the code right now.

Reply:
If a restriction does become apparent, let's create a class for it then.

Also, introducing non-primitive types for primitives will be easier in future, as progress on refactoring will reduce the number of lines of code that will need updating.